JesusFilm/core
# refactor: journeys templates view

# Description

**Note: only merge after  https://github.com/JesusFilm/core/pull/2912!**

### Issue

Components needed for the resources hub need refactored out into the shared space to be used by both journeys-admin and watch. 

We need to refactor out TemplateView components and the related data.ts files that are needed to create journeys pages in the resources hub section of the watch project.

[Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/37663178/todos/7452920587)

### Solution

This PR refactors the TemplateView components into the journeys/ui/ space. 

# External Changes

n/a

# Additional information

Big-bang discovery branch is here https://github.com/JesusFilm/core/pull/2810.

## How To Test
There should be no behavioural impact so ensure application still performs as expected. These components are currently only being used in journeys-admin so regression test as appropriate.
One should be able to use the TemplateView and child components in another project.
